consensus/types/src/data_column_sidecar.rs

```diff
@@ -95,23 +94,32 @@ impl<T: EthSpec> DataColumnSidecar<T> {
         for blob in blobs {
             let blob = KzgBlob::from_bytes(&blob.blob).map_err(KzgError::from)?;
             let (blob_cells, blob_cell_proofs) = kzg.compute_cells_and_proofs(&blob)?;
-            for ((column, column_proofs), (cell, proof)) in
-                iter::zip(columns.iter_mut(), column_kzg_proofs.iter_mut()).zip(iter::zip(
-                    blob_cells.into_iter(),
-                    blob_cell_proofs.into_iter(),
-                ))
-            {
+
+            // we iterate over each column, and we construct the column from "top to bottom",
+            // pushing on the cell and the corresponding proof at each column index. we do this for
+            // each blob (i.e. the outer loop).
+            for col in 0..T::number_of_columns() {
```

Comment:
was this change for addressing the stack overflow, or just style preference?

Reply:
it was to address the stack overflow for the `debug` test profile, but unfortunately we lose the niceness of the iterators and have to index into the vectors.

Reply:
yeah i see, that's unfortunate - i do think the previous approach was nicer too!

Reply:
Nice work, tests are passing! I'm going to address the other minor comment and will merge this ☺️